### PR TITLE
Expo update to SDK 40

### PR DESCRIPTION
**Pull request by petruspierre**
## Summary
Expo SDK 36 have been deprecated since December 9th, so this PR will resolve that. Many packages were updated, so many adjustments had to be made.

## Changelogs
- Expo packages were updated using CLI command `expo upgrade`. Now using [Expo SDK 40](https://blog.expo.io/expo-sdk-40-is-now-available-d4d73e67da33?gi=fa807b0055a5)
- The new packages generated a error on ESLint, so `import/namespace` rule was deactivated and `import/no-cycle` removed from .eslintrc - [Related issue](https://github.com/typescript-eslint/typescript-eslint/issues/1333#issuecomment-573332830)
- Updated testing libraries and jest config and ts config files
- Updated tests snapshots using `jest -u`
- Changed `fireEvent` syntax on Switch component unit test [API Documentation](https://callstack.github.io/react-native-testing-library/docs/api/#fireeventeventname)

### CODE REVIEW

**Review by belgamo — APPROVED** [review]
Great PR! Thanks.

**Comment by belgamo** [comment]
@renanmav can you check why the pipeline wasn't triggered?

File: packages/eslint/index.js [hunk 1]
```diff
@@ -47,7 +47,6 @@ module.exports = {
         ],
       },
     ],
-    'import/no-cycle': 'error',
```
Comment (renanmav) [inline packages/eslint/index.js:50]: Why this was removed?

File: packages/ui/core/Button/__tests__/__snapshots__/Button.test.tsx.snap [hunk 2]
```diff
@@ -1,19 +1,22 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Button renders correctly 1`] = `
-<TouchableOpacity
-  activeOpacity={0.2}
-  gradient={true}
-  onPress={[MockFunction onPress]}
-  rounded={true}
-  size="small"
+<View
```
Comment (renanmav) [inline packages/ui/core/Button/__tests__/__snapshots__/Button.test.tsx.snap:4]: Why is this not an `TouchableOpacity` anymore?

**Review by renanmav — COMMENTED** [review]
For dependency bump in a monorepo, verify that every package that uses dependency A its on the same version across packages.

File: packages/ui/core/Button/__tests__/__snapshots__/Button.test.tsx.snap (hunk 2 above)
Comment (petruspierre) [reply packages/ui/core/Button/__tests__/__snapshots__/Button.test.tsx.snap:4]: It was made automatically using `yarn jest -u` to update the snapshots. This change is related to this issue https://github.com/callstack/react-native-testing-library/issues/450

File: packages/eslint/index.js (hunk 1 above)
Comment (petruspierre) [reply packages/eslint/index.js:50]: To solve the following error:
image
It is related to this issue https://github.com/typescript-eslint/typescript-eslint/issues/1333#issuecomment-566036856

**Pull request merged**